### ISSUE TITLE
Cannot build Microgateway with toolkit "Something wrong with connection"

### ISSUE THREAD

**Issue by McMlok** [wso2/product-microgateway#1318]
### Description:
I try to build a microgateway with the toolkit command micro-gw build. I end with error "Something wrong with the connection", see log bellow. My computer is behind a corporate proxy and I need to specify the proxy for communication to the internet. I try to modify micro-gw.bat and add Java options -Dhttp(s)Host but without success. Is there any possibility how to setup proxy for micro-gw toolkit?

```
Generating sources...WARN: servers url in the open API 'C:\Users\user\source\Tests\WSO2MicroGW\MicroGateway\api_definitions\swagger.json' does not adhere to pattern '<scheme>://<host>/<basePath>/'. Hence the base path defaults to '/'
[DONE]
Compiling source
        wso2/MicroGateway:3.1.0
error:  connection to the remote repository host failed: Something wrong with the connection
error: wso2/gateway:3.1.0::/C:/src/gateway/services/throttle_event_listener.bal:17:1: cannot resolve module 'wso2/jms'
error:  connection to the remote repository host failed: Something wrong with the connection
error: wso2/gateway:3.1.0::/C:/src/gateway/services/token_revocation_event_listener.bal:17:1: cannot resolve module 'wso2/jms'
error:  connection to the remote repository host failed: Something wrong with the connection
error: wso2/gateway:3.1.0::/C:/src/gateway/utils/throttle_util.bal:18:1: cannot resolve module 'wso2/jms'
error: wso2/gateway:3.1.0::/C:/src/gateway/services/throttle_event_listener.bal:27:33: undefined module 'jms'
................
Creating balos
        target\balo\MicroGateway-2019r3-java8-3.1.0.balo
ballerina: Oh no, something really went wrong. Bad. Sad.
```

ballerina-internal.log:
```
[2020-07-10 12:48:24,581] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at org.wso2.ballerinalang.programfile.PackageFileWriter.writePackage(PackageFileWriter.java:52)
	at org.ballerinalang.packerina.writer.BirFileWriter.writeBIRToPath(BirFileWriter.java:100)
	at org.ballerinalang.packerina.task.CreateBirTask.writeImportBir(CreateBirTask.java:84)
	at org.ballerinalang.packerina.task.CreateBirTask.execute(CreateBirTask.java:60)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:394)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:57)
```

### Steps to reproduce:
Build microgateway when your computer is behind corporate proxy
micro-gw build MicroGateway --docker-image SampleGateway:v1 --docker-base-image wso2/micro-gw:latest

### Affected Product Version:
wso2am-micro-gw-toolkit-windows-3.1.0

### Environment details (with versions):
- OS: Windows 10 1809
- Client: wso2am-micro-gw-toolkit-windows-3.1.0
- Env (Docker/K8s): Docker

**Comment by Rajith90** [comment]
Hi @McMlok ,
During the build we try to download the external dependencies from the ballerina central. It seems like there is an issue when trying to connect to ballerina central via the proxy. Can you try configure the settings.xml file as explained in the doc[1], so connection to ballerina central will be done via your corporate proxy. And also we have fixed this issue in the upcoming 3.2.0 release where we don't need connection to ballerina central during the project build. [2]

[1] - https://ballerina.io/learn/publishing-modules-to-ballerina-central/
[2] - https://github.com/wso2/product-microgateway/issues/1316

**Comment by McMlok** [comment]
Thank you for the link to the documentation. I adjust ballerina settings according to the doc. But build process and with the same error. I start Fiddler (Debugging tool for HTTP requests) and build process want to connect to URL https://api.central.ballerina.io/1.0/modules/ballerinax/java.arrays/*/ and this GET request end with 404. I try the same request in a browser but with the same HTTP 404 result.

**Comment by McMlok** [comment]
I tried to use MGW toolkit 3.0.4 and it worked. I don't know where is deference from 3.1.x in the build process, but I don't see any connection outside my machine.

I'm able to use older toolkit at this time but for future I want to test gRPC then I will look forward to 3.2.x version

**Comment by Rajith90** [comment]
@McMlok , MGW 3.0.x series does not need external dependencies due to older version of ballerina is used. But with MGW 3.1.x series we had this limitation of connecting to ballerina central when building the project. 
But as in my previous reply this is fixed in 3.2.0 version and the beta is already out. Please give it a try also.
I am closing this issue since this is fixed in 3.2.0-beta.
